### ISSUE TITLE
Using Eleventy transforms to render asynchronous content inside Nunjucks macros

### ISSUE THREAD

**Issue by ashur** [ashur/multiline-content#5]
### Body

I’m a big fan of using Nunjucks macros to build pre-rendered components in Eleventy projects. Plopping an element into place is a nice, simple affair:

```njk
{%- from "system/component.njk" import component with context -%}

<aside class="theme-gray--s | sidebar">
	{{ component( "widget-today", {
		date: now,
		scope: "sidebar"
	} ) }}
</aside>
```

And unlike `include`, values are passed directly to the macro and don’t depend on variables that potentially pollute the global scope.

There is [a frustrating limitation](https://mozilla.github.io/nunjucks/templating.html#macro), however:

> **Important note**: […] please be aware that you cannot do anything asynchronous inside macros.

In other words, shortcodes that perform asynchronous operations — like optimizing remote images using the [Eleventy Image plugin](https://www.11ty.dev/docs/plugins/image/#asynchronous-shortcode) — will simply render an empty value when called from inside a macro:

```njk
{% macro render( cardImage, description, title, url ) %}
<div class="card">
	{% image cardImage.alt, cardImage.src %}
	<h2 class="card__title">{{ title }}</h2>
	<p class="card__description">{{ description }}</p>
</div>
{% endmacro %}
```
